contracts/adapters/KycOnboarding.sol

```diff
@@ -0,0 +1,267 @@
+        keccak256("kyc-onboarding.signerAddress");
+
+    bytes32 constant ChunkSize = keccak256("kyc-onboarding.chunkSize");
+    bytes32 constant UnitsPerChunk = keccak256("kyc-onboarding.unitsPerChunk");
+    bytes32 constant MaximumChunks = keccak256("kyc-onboarding.maximumChunks");
+    bytes32 constant MaximumUnits =
+        keccak256("kyc-onboarding.maximumTotalUnits");
+    bytes32 constant MaxMembers = keccak256("kyc-onboarding.maxMembers");
+    bytes32 constant FundTargetAddress =
+        keccak256("kyc-onboarding.fundTargetAddress");
+
+    uint256 private _chainId;
+    WETH private _weth;
+    IERC20 private _weth20;
+
+    event CouponRedeemed(
+        address daoAddress,
+        uint256 nonce,
+        address authorizedMember,
+        uint256 amount
+    );
+
+    struct OnboardingDetails {
+        uint88 chunkSize;
+        uint88 numberOfChunks;
+        uint88 unitsPerChunk;
+        uint88 unitsRequested;
+        uint88 maximumTotalUnits;
+        uint160 amount;
+    }
+
+    mapping(DaoRegistry => uint256) public totalUnits;
+
+    constructor(uint256 chainId, address payable weth) {
+        _chainId = chainId;
+        _weth = WETH(weth);
+        _weth20 = IERC20(weth);
+    }
+
+    /**
+     * @notice default fallback function to prevent from sending ether to the contract
+     */
+    receive() external payable {
+        revert("fallback revert");
+    }
+
+    /**
+     * @notice Configures the Adapter with the coupon signer address and token to mint.
+     * @param dao the dao to configure
+     * @param signerAddress is the DAO instance to be configured
+     * @param chunkSize how many wei per chunk
+     * @param unitsPerChunk how many units do we get per chunk
+     * @param maximumChunks  how many chunks can you get at most
+     */
+    function configureDao(
+        DaoRegistry dao,
+        address signerAddress,
+        uint256 chunkSize,
+        uint256 unitsPerChunk,
+        uint256 maximumChunks,
+        uint256 maxUnits,
+        uint256 maxMembers,
+        address fundTargetAddress
+    ) external onlyAdapter(dao) {
+        require(
+            chunkSize > 0 && chunkSize < type(uint88).max,
+            "chunkSize::invalid"
+        );
+        require(
+            maxMembers > 0 && maxMembers < type(uint88).max,
+            "maxMembers::invalid"
+        );
+        require(
+            maximumChunks > 0 && maximumChunks < type(uint88).max,
+            "maximumChunks::invalid"
+        );
+        require(
+            maxUnits > 0 && maxUnits < type(uint88).max,
+            "maxUnits::invalid"
+        );
+        require(
+            unitsPerChunk > 0 && unitsPerChunk < type(uint88).max,
+            "unitsPerChunk::invalid"
+        );
+        require(
+            maximumChunks * unitsPerChunk < type(uint88).max,
+            "potential overflow"
+        );
+
+        require(signerAddress != address(0x0), "signer address is nil!");
```

Comment:
```suggestion
        require(DaoHelper.isNotZeroAddress(signerAddress), "invalid signer address");
```

Reply:
I don't think we really have the nil concept in solidity, just suggested a change to use the DaoHelper and other error msg.